packages/abstraction/test/ContractGenerator.test.ts

```diff
@@ -1,3 +1,98 @@
+      fields: [{
+        name: { type: String, required: true, length: { min: 1, max: 12 }, use: { checkName } },
+        type: { type: String, required: true, use: { checkType } },
+      }],
+      key: { type: String, required: true, length: { min: 1, max: 12 }, use: { checkName } }
+    }
+    ]
+  }
+
+  beforeEach(() => {
+    jest.restoreAllMocks();
+    contractGenerator = new ContractGenerator({
+      aloxideConfigPath: path.resolve(__dirname, './aloxide.yml'),
+      resultPath: path.resolve(__dirname, './out'),
+      adapter,
+      logger
+    });
+  })
+
+  describe('test validateEntity function', () => {
+    it('test validate entity successfully', async () => {
+
+      const validateSchemaMock = jest.spyOn(SchemaValidator, 'validateSchema').mockReturnValueOnce([]);
+      const result = await contractGenerator.validateEntity();
+      expect(result).toBeTruthy();
+      expect(validateSchemaMock).toHaveBeenCalledTimes(1);
+      expect(validateSchemaMock.mock.calls[0][0]).toBe(contractGenerator.aloxideConfig);
+      expect(validateSchemaMock.mock.calls[0][1].toString()).toEqual(requiredSchema.toString());
+      expect(validateSchemaMock.mock.calls[0][2]).toEqual(logger);
+    });
+
+    it('test validate entity failed', async () => {
+      const validateSchemaMock = jest.spyOn(SchemaValidator, 'validateSchema').mockReturnValueOnce([{}, {}]);
+      const result = await contractGenerator.validateEntity();
+      expect(result).toBeFalsy();
+      expect(validateSchemaMock).toHaveBeenCalledTimes(1);
+      expect(validateSchemaMock.mock.calls[0][0]).toBe(contractGenerator.aloxideConfig);
+      expect(validateSchemaMock.mock.calls[0][1].toString()).toEqual(requiredSchema.toString());
+      expect(validateSchemaMock.mock.calls[0][2]).toEqual(logger);
+    });
+  });
+  describe('test generate function', () => {
+    it('test generate but failed cause error validate', async () => {
+      const validateEntityMock = jest.spyOn(contractGenerator, 'validateEntity').mockReturnValueOnce(false);
+      expect(() => { contractGenerator.generate() }).toThrow('input entities mismatch');
+      expect(validateEntityMock).toBeCalledTimes(1);
+    });
+
+    it('test generate but failed cause missing resultPath', async () => {
+      // re-assign value result Path to null
+      contractGenerator.config.resultPath = null;
+      const validateEntityMock = jest.spyOn(contractGenerator, 'validateEntity').mockReturnValueOnce(true);
+      expect(() => { contractGenerator.generate() }).toThrow('missing resultPath');
+      expect(validateEntityMock).toBeCalledTimes(1);
+    })
+    it('test generate success', async () => {
+      const validateEntityMock = jest.spyOn(contractGenerator, 'validateEntity').mockReturnValueOnce(true);
+      try {
+        contractGenerator.generate()
+        expect(validateEntityMock).toBeCalledTimes(1);
+      } catch (error) {
+        expect(error).toBeFalsy();
```

Comment:
How is an `error` falsy in a `catch` block?

Reply:
Cause my expectation this function successful and have no errors

Reply:
I don't understand, if it's successful then it shouldn't throw anything (and `catch` block is never run), do you mean it throw a null value?

@manh-vv @hoang-dao is there anything I missed here?

Reply:
removed